internal/conns/config.go

```diff
@@ -142,9 +142,20 @@ func (c *Config) Client(ctx context.Context) (interface{}, diag.Diagnostics) {
 		return nil, diag.Errorf("error creating AWS SDK v1 session: %s", err)
 	}
 
-	accountID, partition, err := awsbase.GetAwsAccountIDAndPartition(ctx, cfg, &awsbaseConfig)
-	if err != nil {
-		return nil, diag.Errorf("error retrieving account details: %s", err)
+	var accountID string
+	var partition string
+
+	// FIXME: A temporary solution for building correct ARNs while yet resolving via IAM not implemented
+	if c.SkipCredsValidation && c.SkipRequestingAccountId {
+		var parsedAccountID = strings.Split(c.AccessKey, ":")
```

Comment:
```suggestion
		parsedAccountID := strings.Split(c.AccessKey, ":")
```
Советует [styleguide](https://google.github.io/styleguide/go/best-practices.html#initialization)

И кажется, что `parsedAccountID` не совсем корректное название для этой переменной. Может `keyParts` или что-то такое

Reply:
Поправил, вариант `accessKeyParts` мне кажется норм